sunpy/coordinates/transformations.py

```diff
@@ -117,12 +117,9 @@ def hcc_to_hpc(helioccoord, heliopframe):
     Convert from Heliocentic Cartesian to Helioprojective Cartesian.
     """
     if not _observers_are_equal(helioccoord.observer, heliopframe.observer):
-        raise ConvertError("Cannot directly transform heliocentric coordinates to "
-                           "helioprojective coordinates for different "
-                           "observers {} and {}. See discussion in this GH issue: "
-                           "https://github.com/sunpy/sunpy/issues/2712. Try converting to "
-                           "an intermediate heliographic Stonyhurst frame.".format(
-                               helioccoord.observer, heliopframe.observer))
+        heliocframe = Heliocentric(observer=heliopframe.observer)
+        new_helioccoord = hcc_to_hcc(helioccoord,heliocframe)
```

Comment:
Rather than explicitly calling the transformation function here, this should use the transformation machinery:
```python
new_helioccoord = helioccoord.transform_to(heliocframe)
```